File: vessel-market-frontend/src/api/vessels.ts
import { api } from "./http";
import type { Vessel, VesselPayload } from "./types";

export async function getVessels() {
  const { data } = await api.get<Vessel[]>("/vessels/");
  return data;
}

export async function createVessel(payload: VesselPayload) {
  const { data } = await api.post<Vessel>("/vessels/", payload);
  return data;
}

File: backend/app/routers/vessels.py

from fastapi import APIRouter,Depends,HTTPException
from sqlalchemy.orm import Session
from app.db.session import get_db
from app.schemas.vessel import VesselCreate,VesselResponse
from app.services import vessel_service
from app.routers.auth import get_current_user

router=APIRouter(prefix="/vessels", tags=["Vessels"])

@router.post("/", response_model=VesselResponse)
def create(data:VesselCreate, db:Session=Depends(get_db), user=Depends(get_current_user)):
    return vessel_service.create_vessel(db,data,user.id)

@router.get("/", response_model=list[VesselResponse])
def all(db:Session=Depends(get_db)): return vessel_service.get_all(db)
